CompuCell3D/core/CompuCell3D/plugins/MomentOfInertia/MomentOfInertiaPlugin.cpp

```diff
@@ -206,342 +186,97 @@ void CompuCell3D::MomentOfInertiaPlugin::field3DChange(const Point3D &pt, CellG
 string MomentOfInertiaPlugin::toString(){return "MomentOfInertia";}
 
 void MomentOfInertiaPlugin::cellOrientation_xy(const Point3D &pt, CellG *newCell,CellG *oldCell){
-
-	double lMinNew;
-	double lMaxNew;
-	double lMinOld;
-	double lMaxOld;
-
 	if(newCell){
-		double radicalNew=0.5*sqrt((newCell->iXX-newCell->iYY)*(newCell->iXX-newCell->iYY)+4.0*newCell->iXY*newCell->iXY);	
-		lMinNew=0.5*(newCell->iXX+newCell->iYY)-radicalNew;
-		lMaxNew=0.5*(newCell->iXX+newCell->iYY)+radicalNew;
-		//cerr<<"MI lMinOld="<<lMinOld<<" lMaxOld="<<lMaxOld<<endl;
-		double ratio=lMinNew/lMaxNew;
-		if(ratio!=ratio || fabs(ratio)>1.0){
-			newCell->ecc=sqrt(1.0);
-		}
-		else{
-			newCell->ecc=sqrt(1.0-ratio);
-		}
-
-
-		//cerr<<"MI newCell->ecc="<<newCell->ecc<<endl;
-
-		//components of eigenvector corersponding to larger eigenvalue - it points along semiminor axis
-		//in the case of an ellipse larger moment of inertia is w.r.t. semiminor axis and is equal 1/4*a**2*M where a is semimajor axis
-		//double xHat=newCell->iXY; 
-		//double yHat=lMaxNew-newCell->iXX;
-		//double cosAlpha=xHat*xHat/sqrt(xHat*xHat+yHat*yHat);
-		//double sinAlpha=sqrt(1-cosAlpha);
-
-		newCell->lX=newCell->iXY;
-		newCell->lY=lMaxNew-newCell->iXX;
-
-		//if(newCell->iXY!=0.0){
-		//	newCell->lX=newCell->iXY;
-		//	newCell->lY=lMaxOld-newCell->iXX;
-		//}else{
-		//	if(newCell->iXX>newCell->iYY){
-		//		newCell->lX=newCell->iXX;
-		//		newCell->lY=0.0;
-		//	}else{
-
-		//		newCell->lX=0.0;
-		//		newCell->lY=newCell->iYY;
-		//	}
-
-		//}
-
+		auto c = cellOrientation_12(newCell->iXX, newCell->iYY, newCell->iXY);
+		newCell->ecc = (float)c[0];
```

Comment:
I think we can introduce yet another function `assign_orientation_params(Cell Gcell, auto c)` where the 3 assignments below are done in this single function. The only complication would be dealing with lX, lY, lZ etc. So I will leave it as a optional item - if you feel like doing this extra step fine, if not, fine as well we can do it later